pyodide_build/pywasmcross.py

```diff
@@ -33,6 +40,60 @@
     "install_name_tool",
     "otool",
 }
+
+EXCLUDED_LINKER_OPTS = {
+    "-Bsymbolic-functions",
+    # breaks emscripten see https://github.com/emscripten-core/emscripten/issues/14460
+    "--strip-all",
+    "-strip-all",
+    # wasm-ld does not recognize some link flags
+    "--sort-common",
+    "--as-needed",
+    # macOS-specific linker flags that wasm-ld doesn't understand
+    "-headerpad_max_install_names",
+    "-dead_strip_dylibs",
+}
+
+# fmt: off
+EXCLUDED_ARGUMENTS = {
+    # threading is disabled for now
+    "-pthread",
+    # this only applies to compiling fortran code, but we already f2c'd
+    "-ffixed-form",
+    "-fallow-argument-mismatch",
+    # On Mac, we need to omit some darwin-specific arguments
+    "-bundle",
+    "-undefined",
+    "dynamic_lookup",
+    # This flag is needed to build numpy with SIMD optimization which we currently disable
+    "-mpopcnt",
+    # gcc flag that clang does not support
+    "-Bsymbolic-functions",
+    "-fno-second-underscore",
+    "-fstack-protector",  # doesn't work?
+    "-fno-strict-overflow",  # warning: argument unused during compilation
+    "-mno-sse2", # warning: argument unused during compilation
+    "-mno-avx2", # warning: argument unused during compilation
+    "-std=legacy", # fortran flag that clang does not support
+}
+# fmt: on
+
+EXCLUDED_LINKER_PREFIXES = {
+    "--sysroot=",  # ignore unsupported --sysroot compile argument used in conda
+    "--version-script=",
+    "-R/",  # wasm-ld does not accept -R (runtime libraries)
+    "-R.",  # wasm-ld does not accept -R (runtime libraries)
+    "--exclude-libs=",
+}
+
+EXCLUDED_ARG_PREFIXES = {
+    "-J",  # fortran flag that clang does not support
+}
+
+SYS_PREFIX_INCLUDE = sys.prefix + "/include/python"
+SYS_BASE_PREFIX_INCLUDE = sys.base_prefix + "/include/python"
+
+INVOKED_PATH = Path(INVOKED_PATH_STR)
```

Comment:
As above; they are stored as variables. It does hinder navigability, so please let me know and I can move them closer to the methods where they are used.